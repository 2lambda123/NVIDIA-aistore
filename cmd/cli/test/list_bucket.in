ais advanced gen-shards "ais://$BUCKET_1/tmp/shard-{0..4}.tar" --fcount 1 --fsize 1KB --conc 1 --cleanup // IGNORE
ais ls ais://$BUCKET_1
ais ls ais://$BUCKET_1/
ais ls ais://$BUCKET_1
ais ls ais://$BUCKET_1 --props=name,size,version
ais ls ais://$BUCKET_1 --props=name,name,size,size,version
ais ls ais://$BUCKET_1 --props=all
ais ls ais://$BUCKET_1/

ais ls | grep "Buckets" | awk {'print tolower($1)'} | paste -sd "," - // Gathers backend providers from headers.

ais ls s3:// // IGNORE
ais ls aws:// // IGNORE
ais ls aws://$BUCKET_1/ // FAIL "bucket "aws://$BUCKET_1" does not exist"
ais ls az:// // IGNORE
ais ls azure:// // IGNORE
ais ls gs:// // IGNORE
ais ls gcp:// // IGNORE

ais ls #ns // IGNORE
ais ls @ // IGNORE
ais ls @uuid // FAIL "remote cluster "uuid" not found"
ais ls ais://#ns // IGNORE
ais ls ais://#ns/tmp // FAIL "bucket "ais://#ns/tmp" does not exist"
ais ls ais://@uuid#ns // FAIL "remote cluster "uuid" not found"
ais ls ais://@uuid#ns/tmp // FAIL "remote cluster "uuid" not found"

ais ls cloud://$BUCKET_1/ // FAIL "invalid backend provider "cloud""
ais ls aiss://$BUCKET_1 // FAIL "invalid backend provider "aiss""

ais bucket rm ais://$BUCKET_1 // IGNORE
