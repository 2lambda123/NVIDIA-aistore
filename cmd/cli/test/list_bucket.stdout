NAME			 SIZE	 VERSION
tmp/shard-0.tar		 2.50KiB	 1
tmp/shard-1.tar		 2.50KiB	 1
tmp/shard-2.tar		 2.50KiB	 1
tmp/shard-3.tar		 2.50KiB	 1
tmp/shard-4.tar		 2.50KiB	 1
NAME			 SIZE	 VERSION
tmp/shard-0.tar		 2.50KiB	 1
tmp/shard-1.tar		 2.50KiB	 1
tmp/shard-2.tar		 2.50KiB	 1
tmp/shard-3.tar		 2.50KiB	 1
tmp/shard-4.tar		 2.50KiB	 1
NAME			 SIZE	 VERSION
tmp/shard-0.tar		 2.50KiB	 1
tmp/shard-1.tar		 2.50KiB	 1
tmp/shard-2.tar		 2.50KiB	 1
tmp/shard-3.tar		 2.50KiB	 1
tmp/shard-4.tar		 2.50KiB	 1
NAME		 SIZE		 VERSION		 CHECKSUM			 ATIME	 CACHED	 TARGET URL			 STATUS	 COPIES
^tmp/shard-0.tar	 2.50KiB	 1	 (.*)	 .{16}	 yes		 https?://(.*)	 ok	 1$
^tmp/shard-1.tar	 2.50KiB	 1	 (.*)	 .{16}	 yes		 https?://(.*)	 ok	 1$
^tmp/shard-2.tar	 2.50KiB	 1	 (.*)	 .{16}	 yes		 https?://(.*)	 ok	 1$
^tmp/shard-3.tar	 2.50KiB	 1	 (.*)	 .{16}	 yes		 https?://(.*)	 ok	 1$
^tmp/shard-4.tar	 2.50KiB	 1	 (.*)	 .{16}	 yes		 https?://(.*)	 ok	 1$
NAME			 SIZE	 VERSION
tmp/shard-0.tar		 2.50KiB	 1
tmp/shard-1.tar		 2.50KiB	 1
tmp/shard-2.tar		 2.50KiB	 1
tmp/shard-3.tar		 2.50KiB	 1
tmp/shard-4.tar		 2.50KiB	 1
"$BUCKET_1" bucket destroyed
